datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -247,6 +247,14 @@ def create_scraper_configuration(self, instance=None):
         config['type_overrides'] = default_instance.get('type_overrides', {})
         config['type_overrides'].update(instance.get('type_overrides', {}))
 
+        # `_type_override_patterns` is a dictionary where we store Pattern objects
+        # that match metric names as keys, and their corresponding metric type overrrides as values.
+        config['_type_override_patterns'] = {}
+
+        for metric, type in iteritems(config['type_overrides']):
+            if '*' in metric:
+                config['_type_override_patterns'][compile(translate(metric))] = type
```

Comment:
As an extra optimisation that might not be too nasty for the readability, we could delete from `config['types_overrides']` the metrics that have a `*` in their value so that, one type override is either still `config['type_overrides']` (if it has no '*'), or in `config['_type_override_patterns']` (if it has some `*`), but not in the two maps at the same time.

Reply:
will do that 👍